src/parsePref.ts

```diff
@@ -25,6 +25,12 @@ const gitProtocols = new Set([
   'git+ftp',
   'git+file',
   'git+ssh',
+  'http',
+  'https',
+  'rsync',
+  'ftp',
+  'file',
```

Comment:
@huochunpeng
This is why tests started to fail. The file package is currently handled by `https://github.com/pnpm/local-resolver`
Is git via `file:` supported by other package managers?

Reply:
Right, I will get rid of the "file', but retain other 3. Does this sound good?

The reason I add them is that fact npm support them without git+ prefix.

Reply:
http and https may be problematic as well because tarball dependencies may be specified that way.

We can support it as well but it will need some refactoring and more tests in supi to handle all the different edge cases.

Reply:
Ok, I will revert all of them for now.

Reply:
I will leave "ssh" there?

Reply:
It's quite common to use ssh without prefix.

Reply:
ok